docs/07.3-bounded-sendr.html

```diff
@@ -0,0 +1,278 @@
+                    </div>
+
+                    <h1 class="menu-title">Cosock</h1>
+
+                    <div class="right-buttons">
+                        <a href="print.html" title="Print this book" aria-label="Print this book">
+                            <i id="print-button" class="fa fa-print"></i>
+                        </a>
+
+                    </div>
+                </div>
+
+                <div id="search-wrapper" class="hidden">
+                    <form id="searchbar-outer" class="searchbar-outer">
+                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
+                    </form>
+                    <div id="searchresults-outer" class="searchresults-outer hidden">
+                        <div id="searchresults-header" class="searchresults-header"></div>
+                        <ul id="searchresults">
+                        </ul>
+                    </div>
+                </div>
+
+                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
+                <script type="text/javascript">
+                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
+                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
+                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
+                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
+                    });
+                </script>
+
+                <div id="content" class="content">
+                    <main>
+                        <h1 id="bounded-channel-sendr"><a class="header" href="#bounded-channel-sendr">Bounded Channel: Sendr</a></h1>
+<p>To start, we define our Lua table which represents the sending half of our channel.</p>
+<pre><code class="language-lua">local BoundedChannelSender = {}
+BoundedChannelSender.__index = BoundedChannelSender
+</code></pre>
+<p>This table will have the same shape as our <code>BoundedChannelReceiver</code>, it will have a
+<code>_link</code> property and a <code>timeout</code> property; our constructor is nearly identical.</p>
+<pre><code class="language-lua">function BoundedChannelSender.new(shared_queue)
+  return setmetatable({_link = shared_queue}, BoundedChannelSender)
+end
+</code></pre>
+<p>We will also define a <code>settimeout</code> method that looks exactly the same.</p>
+<pre><code class="language-lua">function BoundedChannelSender:settimeout(timeout)
+  self.timeout = timeout
+end
+</code></pre>
+<p>Our <code>close</code> method is also very close to the receiver's implementation.</p>
+<pre><code class="language-lua">function BoundedChannelSender:close()
+  self._link:close()
+  -- If anything is waiting to receieve it should wake up with an error
```

Comment:
Good question, I think it would probably be a good idea but I'll have to think about how that would end up working. In theory all of the senders should be woken at this point which is slightly different than any path we have currently.